connectors/sources/sharepoint_online.py

```diff
@@ -511,6 +536,37 @@ def _validate_sharepoint_rest_url(self, url):
     async def close(self):
         await self._http_session.close()
 
+    async def site_drive_items_delta(self, url):
+        async for response in self._graph_api_client.scroll_with_delta_link(url):
+            delta_link = (
+                response[DELTA_LINK_KEY] if DELTA_LINK_KEY in response else None
+            )
+            if "value" in response and len(response["value"]) > 0:
+                yield DriveItemsPage(response["value"], delta_link)
+
+
+class DriveItemsPage:
```

Comment:
The class itself does not seem specifically attached to Drive Items, so can be named like `DeltaResponsePage`.

On the other hand since it works only for Drive Items now it can be okay to keep it like that. Maybe adding a docstring can help?

Also marking this class with `iterable` and `sized` ABC would be good IMO: https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes

Reply:
Yeah, I'd like to keep it as it is right now. We might change it later when `List Items` API will be available. 

I'll have a look at ABC. Thanks.